### PR TITLE
Implementing services and infra config

### PR DESCRIPTION
**Pull request by MiguelAngelGutierrezMaya**
### Core changes

- Added a new backend service layer with independent Lambda services for `expenses`, `users`, `currencies`, and `documents`, plus a shared package for routing, CORS, error handling, logging, and read/write Postgres access.
- Expanded the expenses domain from config-only support to full CRUD: `GET /expenses` with cursor pagination and `total_count`, `GET /expenses/{id}`, `POST /expenses`, `PUT /expenses/{id}`, `PATCH /expenses/{id}`, `DELETE /expenses/{id}`, and catalog endpoints for `/expenses/types` and `/expenses/categories`.
- Added user and catalog APIs: `POST /users`, `GET /users/{id}`, `PATCH /users/{id}`, `GET /currencies`, and `GET /documents`.
- Added `packages/models` with shared types, JSON schemas, pagination helpers, and standardized error/http-code utilities used by services and API Gateway validation.
- Extended infrastructure with a shared API Gateway, dedicated Lambda stacks for each service, API documentation generation, and Cognito trigger flows for pre-signup provider linking plus post-confirmation/post-authentication user sync.

### Schema changes

- Reworked migrations so `1.0.0` now creates the `readonly_lambda_role`/`readonly_lambda` access setup, and the base relational schema moved to the new major version `2.0.0`.
- The base schema now includes `providers`, expanded `users` fields (`uid`, `identities`, verification flags, `provider_id`), `expenses`, `audit_logs`, `updated_at` triggers, and audit triggers.
- Added RLS policies and grants for both `authenticated` users and the read-only Lambda role, plus seeded catalogs in `2.0.1` and composite indexes for expenses queries in `2.1.0`.

### Minor changes

- Added a GitHub Actions workflow for integration tests, including schema export and isolated test schemas per run.
- Added migration CLI support for `export` and updated env distribution/defaults to cover all services, read-only DB access, API Gateway stage/origins, and integration test fixtures.
- Updated Amplify build config to expose `EXPO_PUBLIC_API_URL` to the client.
- Added broad unit, integration, migration, Cognito, model, and CDK stack test coverage across the new functionality.

## Screenshots/Recordings

N/A

## How to Test

1. Install dependencies with `pnpm install --frozen-lockfile`.
2. Load the required env vars for the new flow (`DATABASE_URL`, `DATABASE_READONLY_URL`, `DB_READONLY_PASSWORD`, `ALLOWED_ORIGINS`, `STAGE`, Cognito vars) and run `pnpm migrations:migrate`. Expected result: migrations finish successfully and the DB includes the new v2 schema, seed data, and read-only role setup.
3. Run `pnpm test` and `pnpm test:integration`. Expected result: unit, infra, migration, and integration suites pass.
4. With the infrastructure deployed in a test environment, authenticate with a valid Cognito user and call `GET /currencies`, `GET /documents`, `GET /expenses/types`, and `GET /expenses/categories`. Expected result: each endpoint returns seeded catalog data for the authenticated user.
5. Create and validate a user profile through `POST /users`, `GET /users/{uid}`, and `PATCH /users/{uid}`. Expected result: the user is created, can be retrieved by UID, and persists profile updates.
6. Validate the expenses flow with `POST /expenses`, `GET /expenses?limit=1`, `GET /expenses/{id}`, `PUT /expenses/{id}`, `PATCH /expenses/{id}`, and `DELETE /expenses/{id}`. Expected result: CRUD operations succeed, the paginated list includes `next_cursor` and `has_more`, and the first page also returns `total_count`.
7. Validate Cognito sync behavior with native and social login/signup flows. Expected result: pre-signup links matching external providers, post-confirmation creates or updates the DB user, and post-authentication keeps the user profile synchronized.

## Checklist

- [x] The PR description includes clear instructions on how to test the implementation manually.

##### Added Tests?

- [x] yes
  - [x] Verified that this PR is not creating duplicate use cases
  - [x] All edge cases described in the acceptance criteria are covered, and tests have clear and descriptive names
- [ ] no, because they aren't needed
- [ ] no, because I need help

_Override these when needed._

```sh

```

### CODE REVIEW

File: services/shared/src/domain/services/logger.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * Generic log attribute object used as a base.
+ */
+type LogAttributes = {
+  [key: string]: unknown;
+};
+
+/**
+ * Log item extra input type.
+ */
+export type LogItemExtraInput = [Error | string] | LogAttributes[];
```
[the remark this answers is not in the document]
Comment (MiguelAngelGutierrezMaya) [reply services/shared/src/domain/services/logger.ts:11]: `@aws-lambda-powertools/logger` is compatible

File: packages/models/src/users/utils.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    phone: data_processed['phone'],
+    document_id: data_processed['document_id'],
+    email_verified: data_processed['email_verified'],
+    phone_verified: data_processed['phone_verified'],
+    provider_id: data_processed['provider_id'],
+    created_at: data_processed['created_at'],
+    updated_at: data_processed['updated_at'],
+    created_by: data_processed['created_by'],
+    modified_by: data_processed['modified_by'],
+  } as UserProfile;
```
[the remark this answers is not in the document]
Comment (MiguelAngelGutierrezMaya) [reply packages/models/src/users/utils.ts:33]: This is a schema validation task

File: services/users/src/application/use-cases/get-user-by-uid.use-case.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import type { UserProfile } from '@packages/models/users/types';
+import type { UserRepository } from '@services/users/domain/repositories/user.repository';
+import { ModuleNotFoundError } from '@packages/models/shared/utils/errors';
+
+export class GetUserByUidUseCase {
+  constructor(private readonly repository: UserRepository) {}
+
+  async execute(uid: string): Promise<UserProfile> {
+    const user = await this.repository.findByUid(uid);
+    if (!user) throw new ModuleNotFoundError();
```
[the remark this answers is not in the document]
Comment (MiguelAngelGutierrezMaya) [reply services/users/src/application/use-cases/get-user-by-uid.use-case.ts:10]: We don't need to worry about this at the moment. This generic error is enough

File: packages/migrations/src/migrations/2/0/0/1_down_drop_base_tables.sql [hunk 4]
```diff
@@ -1,5 +1,10 @@
 SET search_path TO financial_management;
 
+-- Revoke grants from readonly_lambda_role
+REVOKE readonly_lambda_role FROM readonly_lambda;
+REVOKE SELECT ON ALL TABLES IN SCHEMA financial_management FROM readonly_lambda_role;
+REVOKE USAGE ON SCHEMA financial_management FROM readonly_lambda_role;
+
```
[the remark this answers is not in the document]
Comment (MiguelAngelGutierrezMaya) [reply packages/migrations/src/migrations/2/0/0/1_down_drop_base_tables.sql:7]: The migrations are executed in inverse mode

File: services/users/src/router.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  private readonly app: Application;
+  private readonly controller: Controller;
+
+  constructor(app: Application, controller: Controller) {
+    this.app = app;
+    this.controller = controller;
+  }
+
+  static instantiate(app: Application): Router {
+    app.logger.info(`Instantiating router for ${app.pathname}`, Router.name);
```
[the remark this answers is not in the document]
Comment (MiguelAngelGutierrezMaya) [reply services/users/src/router.ts:19]: Test already exist

File: packages/cognito/src/user-sync/application/use-cases/link-existing-providers.use-case.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+   * has NOT been signed up yet. If the social user already exists (was confirmed),
+   * we must delete them first, then link their identity to the native user.
+   *
+   * Flow per social user:
+   * 1. adminDeleteUser(socialUsername) — removes the confirmed social user
+   * 2. adminLinkProviderForUser(native, provider) — links the identity (now unregistered)
+   */
+  async execute(
+    input: LinkExistingProvidersInput,
+  ): Promise<LinkExistingProvidersResult> {
```
[the remark this answers is not in the document]
Comment (MiguelAngelGutierrezMaya) [reply packages/cognito/src/user-sync/application/use-cases/link-existing-providers.use-case.ts:31]: test already exist

File: packages/cognito/src/user-sync/application/use-cases/link-existing-providers.use-case.ts [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+      // 2. Link the provider identity to the native user
+      await this.cognitoAdmin.linkProviderToUser(
+        input.userPoolId,
+        input.nativeUsername,
+        provider.name,
+        provider.userId,
+      );
+      linkedProviders.push(provider.name);
+    }
```
[the remark this answers is not in the document]
Comment (MiguelAngelGutierrezMaya) [reply packages/cognito/src/user-sync/application/use-cases/link-existing-providers.use-case.ts:68]: Test already exist
